### ISSUE TITLE
ghc-8.0.1 incomplete archive for x86_64-linux

### ISSUE THREAD

**Issue by AshleyYakeley** [commercialhaskell/stack#2170]
I set `resolver: ghc-8.0.1` and did `stack setup`:

    $ stack setup
    Preparing to install GHC to an isolated location.
    This will not interfere with any system-level installation.
    Downloaded ghc-8.0.1.                                      
    Running /bin/tar xf /home/ashley/Settings/stack/programs/x86_64-linux/ghc-8.0.1.tar.xz in directory /tmp/stack-setup18672/ exited with ExitFailure 2

    xz: (stdin): Unexpected end of input
    /bin/tar: Unexpected EOF in archive
    /bin/tar: Unexpected EOF in archive
    /bin/tar: Error is not recoverable: exiting now

    Unpacking GHC into /tmp/stack-setup18672/ ...

It looks like the file https://github.com/commercialhaskell/ghc/releases/download/ghc-8.0.1-release/ghc-8.0.1-x86_64-deb7-linux.tar.xz is broken.

**Comment by borsboom** [comment]
Oops, should be fixed now.  Unfortunately it looks like I've hit my large file upload limit on Github for the day, so I've changed it over to S3.

**Comment by AshleyYakeley** [comment]
Thanks

**Comment by sigrlami** [comment]
Got this error for lts 5.15 inside Ubuntu docker, not sure why

**Comment by sigrlami** [comment]
@borsboom 
```
hc-7.10.3:   85.53 MiB / 85.53 MiB (100.00%) downloaded...
Downloaded ghc-7.10.3.
Unpacking GHC into /tmp/stack-setup9/ ...
Running /bin/tar xf /root/.stack/programs/x86_64-linux/ghc-7.10.3.tar.xz in directory /tmp/stack-setup9/ exited with ExitFailure 2

xz: (stdin): Compressed data is corrupt
/bin/tar: Unexpected EOF in archive
```

seeing this also on Fedora 22, 23, 24 and Ubuntu 14.4, 16.4 in Docker, not sure how to fix this

**Comment by borsboom** [comment]
Very strange, it seems to work for me (on Ubuntu 14.04):

```
$ rm -rf ~/.stack/programs/x86_64-linux/ghc-7.10.3*
$ stack setup
Preparing to install GHC to an isolated location.
This will not interfere with any system-level installation.
ghc-7.10.3:   85.53 MiB / 85.53 MiB (100.00%) downloaded...
Downloaded ghc-7.10.3.
Installed GHC.
stack will use a locally installed GHC
For more information on paths, see 'stack path' and 'stack exec env'
To use this GHC and packages outside of a project, consider using:
stack ghc, stack ghci, stack runghc, or stack exec
```

**Comment by sigrlami** [comment]
So, I was testing this almost all day, trying to fix #2311 and it's sometimes works fine and sometimes broke, is that maybe because of S3 outages or something?

**Comment by Tarrasch** [comment]
For what it's worth. I had a similar issue:

```
❯ stack setup
Preparing to install GHC to an isolated location.
This will not interfere with any system-level installation.
ghc-7.10.3:    1.86 MiB / 85.53 MiB (  2.17%) downloaded...     (I manually pressed enter here to save progress of logs)
ghc-7.10.3:   24.00 MiB / 85.53 MiB ( 28.06%) downloaded...
Downloaded ghc-7.10.3.                                     
Running /bin/tar xf /home/arashrk/.stack/programs/x86_64-linux/ghc-7.10.3.tar.xz in directory /tmp/stack-setup26820/ exited with ExitFailure 2


xz: (stdin): Compressed data is corrupt
/bin/tar: Unexpected EOF in archive
/bin/tar: Unexpected EOF in archive
/bin/tar: Error is not recoverable: exiting now

Unpacking GHC into /tmp/stack-setup26820/ ...%                                                                                                                                                              
```

I will try to rerun this.
